src/jdk.incubator.jextract/share/classes/jdk/incubator/jextract/tool/resources/C-X.java.template

```diff
@@ -0,0 +1,86 @@
+// Generated by jextract
+
+import java.lang.invoke.VarHandle;
+import jdk.incubator.foreign.AllocationScope;
+import jdk.incubator.foreign.MemoryAddress;
+import jdk.incubator.foreign.MemoryLayout;
+import jdk.incubator.foreign.MemorySegment;
+
+import static ${C_LANG}.*;
+
+public final class C-X {
+    // don't create!
+    private C-X() {
+    }
+
+    private static VarHandle arrayHandle(MemoryLayout elemLayout, Class<?> elemCarrier) {
+        return MemoryLayout.ofSequence(elemLayout)
+                 .varHandle(elemCarrier, MemoryLayout.PathElement.sequenceElement());
+    }
+
+    public static final MemoryLayout LAYOUT = ${LAYOUT};
+    public static final Class<?> CARRIER = ${CARRIER}.class;
+    private static final VarHandle handle = LAYOUT.varHandle(CARRIER);
+    private static final VarHandle arrayHandle = arrayHandle(LAYOUT, CARRIER);
+
+    public static ${CARRIER} get(MemoryAddress addr) {
+        return (${CARRIER}) handle.get(addr);
+    }
+
+    public static void set(MemoryAddress addr, ${CARRIER} value) {
+        handle.set(addr, value);
+    }
+
+    public static ${CARRIER} get(MemoryAddress addr, long index) {
+        return (${CARRIER}) arrayHandle.get(addr, index);
+    }
+
+    public static void set(MemoryAddress addr, long index, ${CARRIER} value) {
+        arrayHandle.set(addr, index, value);
+    }
```

Comment:
Could maybe name these `getArray` and `setArray`. We also have `allocateArray`, and it would make them a little easier to tell apart from the normal `get` and `set` when reading the code that uses them.

Reply:
Maurizio suggested something similar (adding toJavaArray). We can revisit/adjust later if needed.